src/main/webapp/resources/js/pages/password/PasswordReset.jsx

```diff
@@ -0,0 +1,242 @@
+  const [errorMessages, setErrorMessages] = React.useState(null);
+  const passwordRef = React.useRef();
+
+  /**
+   * When the component gets added to the page,
+   * focus on the password input input.
+   */
+  React.useEffect(() => {
+    passwordRef.current.focus();
+    passwordRef.current.select();
+  }, []);
+
+  const passwordRules = [
+    i18n("PasswordReset.alert.rule2"),
+    i18n("PasswordReset.alert.rule3"),
+    i18n("PasswordReset.alert.rule4"),
+    i18n("PasswordReset.alert.rule5"),
+    i18n("PasswordReset.alert.rule6"),
+    i18n("PasswordReset.alert.recommendation1"),
+    i18n("PasswordReset.alert.recommendation2"),
+  ];
+
+  const submitPasswordResetForm = () => {
+    setLoading(true);
+    setPassword({
+      resetId: passwordResetObj.id,
+      password: passwordResetForm.getFieldValue("password"),
+    }).then((res) => {
+      if (res.error) {
+        setErrorMessages(res.error.data.errors);
+        setUpdateError(true);
+      } else {
+        setUpdateSuccess(true);
+        setUpdateError(false);
+      }
+      passwordResetForm.resetFields();
+      setLoading(false);
+    });
+  };
+
+  // Validator to check if password meets password policy requirements
+  const validatePassword = (password) => {
+    const passwordRegex =
+      /^(?=.*\d)(?=.*[!@#$%^&*()+?/<>={}.\\])(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
+    const minimumPasswordLength = 8;
+    setInvalidPassword(true);
+
+    if (password.length !== 0) {
+      if (password.length >= minimumPasswordLength) {
+        if (passwordRegex.test(password)) {
+          setInvalidPassword(false);
+          return Promise.resolve();
+        } else {
+          return Promise.reject(i18n("PasswordReset.input.passwordNotMatch"));
+        }
+      }
+      return Promise.reject(i18n("PasswordReset.input.minLength"));
+    } else {
+      return Promise.reject(i18n("PasswordReset.passwordIsRequired"));
+    }
+  };
+
+  return (
+    <Row justify="center">
+      <Col style={{ width: 600 }}>
+        <Row justify="center" style={{ marginBottom: SPACE_MD }}>
+          <img
+            src={setBaseUrl("/resources/img/irida_logo_light.svg")}
+            height={60}
+            alt={i18n("generic.irida.website")}
+          />
+        </Row>
+
+        {updateSuccess && (
+          <Alert
+            style={{ marginBottom: SPACE_SM }}
+            message={i18n("password.reset.success")}
+            className="t-reset-success-alert"
+            action={
+              <Button
+                type="link"
+                onClick={() => window.location.replace(setBaseUrl("/"))}
+              >
+                {i18n("password.reset.success.link")}
+              </Button>
+            }
+            type="success"
+            showIcon
+          />
+        )}
+
+        {updateError && (
+          <Alert
+            style={{ marginBottom: SPACE_SM }}
+            className="t-reset-error-alert"
+            message="The following errors occurred:"
```

Comment:
Internationalization

Reply:
Added in eb461f960fdd78a45922a9339cbc750424d85595